src/App.tsx

```diff
@@ -47,28 +47,27 @@ function App() {
 
   useEffect(() => {
     const code = sessionValidationCode.join('')
-    if (code.length === 6 && sessionValidationSalt) {
-      sequence.finishValidateSession(sessionValidationSalt, code)
+    if (code.length === 6) {
       setIsFinishValidateSessionPending(true)
+      sequence.finishValidateSession(code)
     }
   }, [sessionValidationCode])
 
-  sequence.onValidationRequired(salt => {
-    setSessionValidationSalt(salt)
+  sequence.onValidationRequired(() => {
```

Comment:
This should probably be in a `useEffect` with the cleanup callback returned, so that the validation callback is removed when this component unmounts

Reply:
Done